board/safety/safety_hyundai.h

```diff
@@ -145,14 +150,23 @@ static int hyundai_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
       cruise_engaged_prev = cruise_engaged;
     }
 
-    if ((addr == 608) || (hyundai_legacy && (addr == 881))) {
-      if (addr == 608) {
-        gas_pressed = (GET_BYTE(to_push, 7) >> 6) != 0;
+    if ((addr == 881) && (hyundai_hev_gas || hyundai_ev_gas)) {
+      if (hyundai_hev_gas) {
+        gas_pressed = GET_BYTE(to_push, 7) != 0;
       } else {
         gas_pressed = (((GET_BYTE(to_push, 4) & 0x7F) << 1) | GET_BYTE(to_push, 3) >> 7) != 0;
       }
+    } else if (addr == 608) {  // ICE
+      gas_pressed = (GET_BYTE(to_push, 7) >> 6) != 0;
+    } else {
     }
 
+    // exit controls on rising edge of gas press
```

Comment:
this is done here https://github.com/commaai/panda/blob/master/board/safety.h#L204